src/main.rs

```diff
@@ -67,7 +71,20 @@ fn validate_simulation() {
 
 /// Simulate the rabbit population
 fn simulate(starting_rabbits: i128) -> i32 {
-    todo!()
+    let mut count = 0;
+    let mut rabbits = starting_rabbits;
+    loop {
+        if rabbits <= 1{
+            return count;
+        }
+        else if rabbits % 2 != 0{
+            rabbits = (rabbits * 3) + 1;
+        }
+        else{
+            rabbits = rabbits / 2;
```

Comment:
One thing I saw from some other students (that I didn't think to do myself) was to do `rabbits /= 2;` on this line. Kind of neat!